packages/discord.js/src/structures/MessagePayload.js

```diff
@@ -201,6 +201,37 @@ class MessagePayload {
       }
     }
 
+    if (this.options.messageReference) {
+      const reference = this.options.messageReference;
+
+      if (typeof reference === 'object') {
+        const message_id = this.isMessage
+          ? (reference.id ?? reference)
+          : this.target.messages.resolveId(reference.id ?? reference.messageId);
+        const channel_id = this.target.client.channels.resolveId(reference.channelId);
+        const guild_id = this.target.client.guilds.resolveId(reference.guildId);
+
+        if (message_id) {
+          message_reference = {
+            message_id,
+            channel_id,
+            guild_id,
+            type: reference.type,
```

Comment:
One minor concern about my implementation here is if a Message instance is passed into `messageReference`. If that happens, `reference.type` will resolve to the message's type which could be an invalid message reference type. Is this something we want to address or let the end user handle themselves via spreading & overriding syntax or similar?

Reply:
I'd honestly just drop accepting a message. It's easier and no problem to use a message reference object.